File: EJERCICIO_1/EJERCICIOJAVA/discount.java
package EJERCICIOJAVA;

public  abstract class discount {
    protected double percentage; 
    protected salaryBass salary;  

    public discount(salaryBass salary, double percentage) {
        this.salary = salary; 
        this.percentage = percentage;

    }

    public  abstract double calculateDiscount()
     ;

    public static double calculateTotalDiscount(arl arlDiscount, health healthDiscount, pension pensionDiscount) {
        return arlDiscount.calculateDiscount() + healthDiscount.calculateDiscount() + pensionDiscount.calculateDiscount();
    }

    }

File: EJERCICIO_1/EJERCICIOSPHP/discount.php
<?php
abstract class Discount {
    protected $percentage; 
    protected $salaryBass;

    public function __construct($salaryBass, $percentage) {
        $this->salaryBass = $salaryBass; 
        $this->percentage = $percentage;
    }

    public abstract function calculateDiscount();

    public  static function calculateTotalDiscount($arlDiscount, $healthDiscount, $pensionDiscount) {
        return $arlDiscount->calculateDiscount() + $healthDiscount->calculateDiscount() + $pensionDiscount->calculateDiscount();
    }
}
?>

